Popular Movies/Popular Movies/Controllers/MovieDetailTableViewController.m

```diff
@@ -0,0 +1,258 @@
+#import "CastTableViewCell.h"
+#import "MovieDescriptionTableViewCell.h"
+#import "ReviewTableViewCell.h"
+#import <RestKit/RestKit.h>
+#import "VideosCollection.h"
+#import "Trailer.h"
+#import "ActorCollectionViewCell.h"
+#import "ReviewsTableView.h"
+#import "SingleReviewTableViewCell.h"
+#import "ActorCollectionView.h"
+#import "CrewMember.h"
+@interface MovieDetailTableViewController (){
+    MovieService* movieService;
+    NSMutableArray* writers;
+    NSMutableArray* stars;
+    NSMutableArray* director;
+    Movie* selectedMovie;
+    NSString* trailerLink;
+    VideosCollection* videos;
+    CastCollection* castCollection;
+    NSArray* cast;
+    CollectionReview* reviewsCollection;
+    Trailer* t;
+}
+@end
+
+@implementation MovieDetailTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    [self.tableView  registerNib:[UINib nibWithNibName:NSStringFromClass([MovieTrailerTableViewCell class]) bundle:nil] forCellReuseIdentifier:movieTrailerCellReuseIdentifier];
+    [self.tableView  registerNib:[UINib nibWithNibName:NSStringFromClass([ImageGalleryTableViewCell class]) bundle:nil] forCellReuseIdentifier:imageGalleryReuseIdentifier];
+    [self.tableView  registerNib:[UINib nibWithNibName:NSStringFromClass([MovieDescriptionTableViewCell class]) bundle:nil] forCellReuseIdentifier:descriptionReuseIdentifier];
+    [self.tableView  registerNib:[UINib nibWithNibName:NSStringFromClass([CastTableViewCell class]) bundle:nil] forCellReuseIdentifier:castReuseIdentifier];
+    [self.tableView registerNib:[UINib nibWithNibName:NSStringFromClass([ReviewTableViewCell class]) bundle:nil] forCellReuseIdentifier:reviewReuseIdentifier];
+    
+    movieService = [[MovieService alloc]init];
+    [self loadMovieDetails];
+    [self loadMovieTrailerByMovieId];
+    [self loadCastDetails];
+    [self loadReviews];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    if([tableView isKindOfClass:[ReviewsTableView class]]) return 1;
+    return 4;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if([tableView isKindOfClass:[ReviewsTableView class]]){
+        return reviewsCollection.results.count;
+    }
+    return 1;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell* cell = [[UITableViewCell alloc]init];
+    if([tableView isKindOfClass:[ReviewsTableView class]]){
+        SingleReviewTableViewCell *reviewCell = (SingleReviewTableViewCell*)[tableView dequeueReusableCellWithIdentifier:singleReviewReuseIdentifier forIndexPath:indexPath];
+        SingleReview* singleReview = [[SingleReview alloc]init];
+        singleReview = (SingleReview*)[reviewsCollection.results objectAtIndex:indexPath.row];
+        reviewCell.nameLabel.text = singleReview.author;
```

Comment:
Good approach here is that you create single method in <Some>Cell class e.g setupSingleReviewCell which should have input parameters depending on all data you need to pass in cell e.g author and content. Then you should make labels private attributes and modify them in that method.

E.g.
(void)setupSingleReviewCellWithAuthor:(NSString*)author andContent(NSString*)content {
    self.namelabel.text = author;
    self.reviewContentLabel.text = content;
}